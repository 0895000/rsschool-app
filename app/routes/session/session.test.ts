import { sessionMiddleware } from './index';
import { IRouterContext } from 'koa-router';

describe('Session', () => {
    it('returns user session', async () => {
        const ctx: IRouterContext = {
            state: {
                user: {
                    _id: 'apalchys',
                },
            },
        } as any;
        await sessionMiddleware(ctx);
        expect(ctx.body).toMatchSnapshot();
        expect(ctx.status).toBe(200);
    });

    it('returns 404 if no user session', async () => {
        const ctx: IRouterContext = {
            state: {},
        } as any;
        await sessionMiddleware(ctx);
        expect(ctx.body).toMatchSnapshot();
        expect(ctx.status).toBe(404);
    });
});
