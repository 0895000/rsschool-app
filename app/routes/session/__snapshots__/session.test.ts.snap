// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Session returns 404 if no user session 1`] = `undefined`;

exports[`Session returns user session 1`] = `
Object {
  "data": Object {
    "_id": "apalchys",
  },
}
`;
